pkg/parse/parse.go

```diff
@@ -570,21 +570,30 @@ func SystemContextFromOptions(c *cobra.Command) (*types.SystemContext, error) {
 			ctx.VariantChoice = variant
 		}
 	}
+	var platform string
 	if c.Flag("platform") != nil && c.Flag("platform").Changed {
-		if platform, err := c.Flags().GetString("platform"); err == nil {
-			os, arch, variant, err := Platform(platform)
-			if err != nil {
-				return nil, err
-			}
-			if ctx.OSChoice != "" ||
-				ctx.ArchitectureChoice != "" ||
-				ctx.VariantChoice != "" {
-				return nil, errors.Errorf("invalid --platform may not be used with --os, --arch, or --variant")
-			}
-			ctx.OSChoice = os
-			ctx.ArchitectureChoice = arch
-			ctx.VariantChoice = variant
+		var specs []string
+		if specs, err = c.Flags().GetStringSlice("platform"); err != nil {
+			return nil, err
+		}
+		if len(specs) == 0 || specs[0] == "" {
+			return nil, errors.Errorf("unable to parse --platform value %v", specs)
 		}
+		platform = specs[0]
+	}
+	if platform != "" {
```

Comment:
Is there a reason to make this a separate conditional?

Entirely up to you — just noticing that this doesn’t look like the prevalent coding style.

Reply:
That's a leftover from an earlier revision.  Merging them.